irohad/main/impl/subscription.cpp

```diff
@@ -6,13 +6,23 @@
 #include "main/subscription.hpp"
 
 #include <fstream>
+#include <mutex>
 
 namespace iroha {
 
   std::shared_ptr<Subscription> getSubscription() {
-    static std::shared_ptr<Subscription> engine =
-        std::make_shared<Subscription>();
-    return engine;
+    static std::weak_ptr<Subscription> engine;
```

Comment:
why first try without mutex?

Reply:
Because it doesn't needed here. ```lock``` is a safe call.